ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -298,25 +299,31 @@ public static String buildAvroSchema(final Schema schema, String name) {
   private static org.apache.avro.Schema getAvroSchemaForField(Schema fieldSchema) {
     switch (fieldSchema.type()) {
       case STRING:
-        return create(org.apache.avro.Schema.Type.STRING);
+        return unionWithNull(create(org.apache.avro.Schema.Type.STRING));
       case BOOLEAN:
-        return create(org.apache.avro.Schema.Type.BOOLEAN);
+        return unionWithNull(create(org.apache.avro.Schema.Type.BOOLEAN));
       case INT32:
-        return create(org.apache.avro.Schema.Type.INT);
+        return unionWithNull(create(org.apache.avro.Schema.Type.INT));
       case INT64:
-        return create(org.apache.avro.Schema.Type.LONG);
+        return unionWithNull(create(org.apache.avro.Schema.Type.LONG));
       case FLOAT64:
-        return create(org.apache.avro.Schema.Type.DOUBLE);
+        return unionWithNull(create(org.apache.avro.Schema.Type.DOUBLE));
       default:
         if (fieldSchema.type() == Schema.Type.ARRAY) {
-          return createArray(getAvroSchemaForField(fieldSchema.valueSchema()));
+          return unionWithNull(
+              createArray(getAvroSchemaForField(fieldSchema.valueSchema())));
         } else if (fieldSchema.type() == Schema.Type.MAP) {
-          return createMap(getAvroSchemaForField(fieldSchema.valueSchema()));
+          return unionWithNull(
+              createMap(getAvroSchemaForField(fieldSchema.valueSchema())));
         }
         throw new KsqlException("Unsupported AVRO type: " + fieldSchema.type().name());
     }
   }
 
+  private static org.apache.avro.Schema unionWithNull(org.apache.avro.Schema schema) {
+    return createUnion(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.NULL), schema);
```

Comment:
still think we should set the default value for the field to null

Reply:
Made default value `null`.